lib/Wikia/src/Service/User/Attributes/AttributeKeyValueService.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Wikia\Service\User\Attributes;
+
+use Wikia\Logger\Loggable;
+use Wikia\Persistence\User\Attributes\AttributePersistence;
+use Wikia\Util\WikiaProfiler;
+use Wikia\Service\User\Attributes;
+
+class AttributeKeyValueService implements AttributeService {
+
+	const PROFILE_EVENT = \Transaction::EVENT_USER_ATTRIBUTES;
+
+	use WikiaProfiler;
+	use Loggable;
+
+	/** @var AttributePersistence */
+	private $persistenceAdapter;
+
+	/**
+	 * @Inject({
+	 *    Wikia\Persistence\User\Attributes\AttributePersistence::class
+	 * })
+	 * @param AttributePersistence $persistenceAdapter
+	 */
+	function __construct( AttributePersistence $persistenceAdapter ) {
+		$this->persistenceAdapter = $persistenceAdapter;
+	}
+
+	public function set( $userId, $attribute ) {
+		if ( empty( $attribute ) || $userId === 0 ) {
+			return false;
```

Comment:
Yep, that definitely got missed :) Sure, that works too. A lot of this code was heavily inspired by the User Preferences Service (read, borderline plagiarized), so this was a hold over from that. An exception sounds good there. I'll add it.